src/components/builder.ts

```diff
@@ -126,7 +126,7 @@ export class Builder {
                 args: configuration.get('latex.magiccommand.args') as string[]
             }]
         } else {
-            const recipes: {name: string, tools: string[]}[] = configuration.get('latex.recipes') as {name: string, tools: string[]}[]
+            const recipes = configuration.get('latex.recipes') as {name: string, tools: (string | StepCommand)[]}[]
```

Comment:
Can you please elaborate on why `tools` field can be a string?

Reply:
I think it would be better to allow *raw* StepCommand as `latex-workshop.latex.recipes[n].tools` element (it is allowed in my original implementation). If you don't want to do this, I'll rollback it (and also delete line.156).

Reply:
I would suggest a uniformed definition of `tools`. This can be somewhat annoying at first, but multi-recipe users should be happy to see a tidy configuration file.

Reply:
Maybe let it be a hidden undocumented feature would work well.

Reply:
Of course I think it is not a clean solution.
Pls remove it later when you think it is unnecessary or someone implements good alternatives.